node/node.go

```diff
@@ -797,18 +797,21 @@ func (app *App) initServices(ctx context.Context, poetClients []activation.PoetP
 		app.log.Panic("failed to create post setup manager: %v", err)
 	}
 
-	nipostBuilder := activation.NewNIPostBuilder(
+	nipostBuilder, err := activation.NewNIPostBuilder(
 		app.edSgn.NodeID(),
 		postSetupMgr,
-		poetClients,
 		poetDb,
+		app.Config.PoETServers,
 		app.Config.SMESHING.Opts.DataDir,
 		app.addLogger(NipostBuilderLogger, lg),
 		app.edSgn,
 		poetCfg,
 		app.clock,
 		activation.WithNipostValidator(app.validator),
 	)
+	if err != nil {
+		app.log.Panic("failed to create nipost builder: %v", err)
+	}
```

Comment:
Too bad error handling is mixed (sometimes a panic, sometimes bubble up).

Reply:
I went for panic because the other services immediately proceeding also panic on error. It also seems like the correct response to me (without the service being able to be initialized it doesn't make sense to continue and just exist with the error).